SensorArray.cpp

```diff
@@ -0,0 +1,94 @@
+#include "SensorArray.h"
+#include "Sensor.h"
+
+
+using namespace std;
+
+namespace LFRobot
+{
+	SensorArray::SensorArray(int pins[])
+	{
+		Sensor* sensorArray[NUMBER_OF_SENSORS];
```

Comment:
This is a re-declaration of the member variable at SensorArray.h:23. C++ allows this by hiding (shadowing) the variable with the wider scope ([read more about it here](https://en.wikipedia.org/wiki/Variable_shadowing)). So what this means is that when you assign to `sensorArray` on line 5 you are assigning to the local one, not the class level one. So basically a major memory leak when `sensorArray` goes out of scope at the end of the constructor. Every other use of `sensorArray` in the class is using the un-initialized one (bad).


On another note, I know you just learned about pointers, so is good time to learn when pointers are not needed. Notice in your implementation you never move a pointer around, reassign it, pass it to a function etc. You aren't using many of the benefits of pointers but are still stuck with all the negatives (null dereference, memory leak, etc.). The one benefit you are using is that you can delay creating the objects in the array until you are able to put meaningful objects in it, but this is (generally)[https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c43-ensure-that-a-class-has-a-default-constructor] not considered a good enough reason to use pointers.